src/main/java/dev/binclub/javaception/classfile/MethodInfo.java

```diff
@@ -8,19 +9,22 @@
 	public final int access;
 	public final String name;
 	public final Type[] descriptor;
+	public final String signature;
 	public CodeAttribute code;
+	public Klass owner;
+	private int _hashcode = 0;
 	
 	public MethodInfo(int access, String name, String descriptor) {
-		this(access, name, Type.parseMethodDescriptor(descriptor));
+		this(access, name, Type.parseMethodDescriptor(descriptor), descriptor);
 	}
 	
-	public MethodInfo(int access, String name, Type[] descriptor) {
+	public MethodInfo(int access, String name, Type[] descriptor, String signature) {
```

Comment:
I don't think we need to retain signature info

Reply:
signature is kept so arrays dont have to be iterated to see if methods are equal

Reply:
Should be called something like `descriptorStr` not signature, signature stores the generic types not the descriptor.